core/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -290,12 +291,14 @@ protected void doRun() throws Exception {
                         case INDEX:
                             IndexRequest indexRequest = (IndexRequest) docWriteRequest;
                             MappingMetaData mappingMd = null;
+                            Version indexCreated = null;
                             final IndexMetaData indexMetaData = metaData.index(concreteIndex);
                             if (indexMetaData != null) {
```

Comment:
IMO this can never happen. if you at what we do [here](https://github.com/jpountz/elasticsearch/blob/3fe833d1742db162db47f2c13969649090b70a24/core/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java#L284) we add a failure and skip this entire block when then index metadata is not available. I think this is legacy and can / must be removed. I still think @bleskes should confirm that since he knows this class best